modules/programs/yt-dlp.nix

```diff
@@ -0,0 +1,47 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.yt-dlp;
+
+in {
+  meta.maintainers = [ maintainers.marsam ];
+
+  options.programs.yt-dlp = {
+    enable = mkEnableOption "yt-dlp";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.yt-dlp;
+      defaultText = literalExpression "pkgs.yt-dlp";
+      description = "Package providing the <command>yt-dlp</command> tool.";
+    };
+
+    settings = mkOption {
```

Comment:
I would suggest renaming this to `extraConfig` since `settings` typically is reserved for structured configuration à la RFC 42.

Reply:
Updated. Thanks!